utils/build-script

```diff
@@ -1473,6 +1514,191 @@ details of the setups of other systems or automated environments.""")
+    of options.
+    """
+
+    # Compute the platform to build for information.
+    platforms_to_skip_build = set()
+    if args.skip_build_linux:
+        platforms_to_skip_build.add(StdlibDeploymentTarget.Linux)
+    if args.skip_build_freebsd:
+        platforms_to_skip_build.add(StdlibDeploymentTarget.FreeBSD)
+    if args.skip_build_cygwin:
+        platforms_to_skip_build.add(StdlibDeploymentTarget.Cygwin)
+    if args.skip_build_osx:
+        platforms_to_skip_build.add(StdlibDeploymentTarget.OSX)
+    if args.skip_build_ios_device:
+        platforms_to_skip_build.add(StdlibDeploymentTarget.iOS)
+    if args.skip_build_ios_simulator:
+        platforms_to_skip_build.add(StdlibDeploymentTarget.iOSSimulator)
+    if args.skip_build_tvos_device:
+        platforms_to_skip_build.add(StdlibDeploymentTarget.AppleTV)
+    if args.skip_build_tvos_simulator:
+        platforms_to_skip_build.add(StdlibDeploymentTarget.AppleTVSimulator)
+    if args.skip_build_watchos_device:
+        platforms_to_skip_build.add(StdlibDeploymentTarget.AppleWatch)
+    if args.skip_build_watchos_simulator:
+        platforms_to_skip_build.add(StdlibDeploymentTarget.AppleWatchSimulator)
+    if args.skip_build_android:
+        platforms_to_skip_build.add(StdlibDeploymentTarget.Android)
+
+    platforms_to_skip_test = set()
+    if args.skip_test_linux:
+        platforms_to_skip_test.add(StdlibDeploymentTarget.Linux)
+    if args.skip_test_freebsd:
+        platforms_to_skip_test.add(StdlibDeploymentTarget.FreeBSD)
+    if args.skip_test_cygwin:
+        platforms_to_skip_test.add(StdlibDeploymentTarget.Cygwin)
+    if args.skip_test_osx:
+        platforms_to_skip_test.add(StdlibDeploymentTarget.OSX)
+    if args.skip_test_ios_host:
+        platforms_to_skip_test.add(StdlibDeploymentTarget.iOS)
+    if args.skip_test_ios_simulator:
+        platforms_to_skip_test.add(StdlibDeploymentTarget.iOSSimulator)
+    if args.skip_test_tvos_host:
+        platforms_to_skip_test.add(StdlibDeploymentTarget.AppleTV)
+    if args.skip_test_tvos_simulator:
+        platforms_to_skip_test.add(StdlibDeploymentTarget.AppleTVSimulator)
+    if args.skip_test_watchos_host:
+        platforms_to_skip_test.add(StdlibDeploymentTarget.AppleWatch)
+    if args.skip_test_watchos_simulator:
+        platforms_to_skip_test.add(StdlibDeploymentTarget.AppleWatchSimulator)
+
+    platforms_to_skip_test_host = set()
+    if args.skip_test_ios_host:
+        platforms_to_skip_test_host.add(StdlibDeploymentTarget.iOS)
+    if args.skip_test_tvos_host:
+        platforms_to_skip_test_host.add(StdlibDeploymentTarget.AppleTV)
+    if args.skip_test_watchos_host:
+        platforms_to_skip_test_host.add(StdlibDeploymentTarget.AppleWatch)
+
+    # We never allow testing Android, currently.
+    #
+    # FIXME: Allow Android host tests to be enabled/disabled by the build
+    # script.
+    platforms_to_skip_test.add(StdlibDeploymentTarget.Android)
+
+    def get_options_for_target(host):
```

Comment:
I agree, this part is the "WIP" part, and I want to find a way to clean all this code up more and host it in a nicer location.